src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -128,6 +128,107 @@ inline void ResizeImages(arma::Mat<eT>& images, data::ImageInfo& info,
   info.Height() = newHeight;
 }
 
+/**
+ * Resize & Crop one single image matrix or a set of images.
+ *
+ * This function should be used if the image is loaded as an armadillo matrix
+ * and the number of cols equal to the Width and the number of rows equal
+ * the Height of the image, or the total number of image pixels is equal to the
+ * number of element in an armadillo matrix.
+ *
+ * The same applies if a set of images is loaded, but all of them need to have
+ * identical dimension when loaded to this matrix.
+ *
+ * @param image The input matrix that contains the image to be resized.
+ * @param info Contains relevant input images information.
+ * @param newWidth The new requested width for the resized image.
+ * @param newHeight The new requested height for the resized image.
+ */
+template<typename eT>
+inline void CropResizeImages(arma::Mat<eT>& images, data::ImageInfo& info,
+    const size_t newWidth, const size_t newHeight)
+{
+  float ratioW = static_cast<float>(newWidth)  / static_cast<float>(info.Width());
+  float ratioH = static_cast<float>(newHeight) / static_cast<float>(info.Height());
+
+  float largestRatio = ratioW > ratioH ? ratioW : ratioH;
+  int midWidth = static_cast<int>(largestRatio * info.Width());
+  int midHeight = static_cast<int>(largestRatio * info.Height());
+
+  // Edge cases, what if the width / height value is odd ? then increase the
+  // resize value to the closest pair.
+  if (midHeight % 2 != 0 && midWidth % 2 == 0)
+      midHeight = midHeight + 1; 
+  else if (midWidth % 2 !=0 && midHeight % 2 == 0)
+    midWidth = midWidth + 1;
+  else if (midWidth % 2 !=0 && midHeight % 2 != 0)
+  {
+    midHeight = midHeight + 1;
+    midWidth = midWidth + 1;
+  }
+
+  ResizeImages(images, info, midWidth, midHeight);
```

Comment:
I'm not sure I understand why we are calling `ResizeImages()` here; that function will rescale the image to the new size, but I thought the function here was supposed to just crop and center the image, but not transform the pixels at all?

If you can add a before/after example (or two :smile:) to the documentation, then I think I will be able to understand what the function is supposed to do a lot better and can give a better review.

Reply:
Yes, it is Resizing and cropping, the idea is to respect the aspect ratio, especially if you have two different dimensions.
I need to add an example to make it easier to understand.

Reply:
Is it true that the order does not matter here?  Unless I am misunderstanding, I think it is equivalent to crop first and resize second, vs. resizing first and cropping second.  (The sizes you would use for the resize in each case would be different.)

I'm asking this to check my understanding of what's going on here---if that is correct, then I think maybe keeping `ResizeImages()` a separate step is a good idea for the sake of simplicity.